### PR TITLE
Dynamic Filter Form Improvement

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #1998 
# What's Changed

- Instead of a human-friendly content-type display name, the lookup value for ContentTypes displays integers. e.g 
**Before**
```html
<select ..>
    <option value="dcim.site">1</option>
    <option value="dcim.device">2</option>
    ...
</select>
```
**Now**
```html
<select ..>
    <option value="dcim.site">dcim | site</option>
    <option value="dcim.device">dcim | device</option>
    ...
</select>
```

- Remove the whitespace to the right of the delete button on the DynamicFilterForm modal.
Screenshot 2022-10-17 at 4 37 48 PM

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/utils.py [hunk 1]
```diff
@@ -165,12 +165,11 @@ def __call__(self):
 
         return query
 
-    @property
```
Comment (timizuoebideri1) [inline nautobot/extras/utils.py:168]: To make `TaggableClassesQuery.as queryset` consistent with both `ChangeLoggedModelsQuery.as queryset` and `FeatureQuery.as queryset`, `@property` was removed.
`as_queryset` is required to be consistent here, therefore it has to be done.
```python
queryset_map = {"tags": TaggableClassesQuery, "job hooks": ChangeLoggedModelsQuery}
form_field = MultipleContentTypeField( choices_as_strings=True, queryset=queryset_map[plural_name]().as_queryset() )
```

**Review by HanlinMiao — APPROVED** [review]
LGTM!

File: nautobot/utilities/utils.py [hunk 2]
```diff
@@ -800,6 +802,18 @@ def get_filterset_parameter_form_field(model, parameter):
...
+        plural_name = model._meta.verbose_name_plural
+        try:
+            form_field = MultipleContentTypeField(choices_as_strings=True, feature=plural_name)
+        except KeyError:
+            # `MultipleContentTypeField` employs `registry["model features"][feature]`, which may
+            # result in an error if `feature` is not found in the `registry["model features"]` dict.
+            # In this case use queryset
+            queryset_map = {"tags": TaggableClassesQuery, "job hooks": ChangeLoggedModelsQuery}
+            form_field = MultipleContentTypeField(
+                choices_as_strings=True, queryset=queryset_map[plural_name]().as_queryset()
```
[the remark this answers is not in the document]
Comment (timizuoebideri1) [reply nautobot/utilities/utils.py:815]: No, as far as I can see, this particular case handling would only be required for `tags` and `job hooks.`

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
Is this still a draft, and if so what's needed to get it ready to go? If it's ready to go already, should we merge it?

**Pull request merged**